datalad_container/containers_run.py

```diff
@@ -131,14 +132,41 @@ def __call__(cmd, container_name=None, dataset=None,
             # just prepend and pray
             cmd = container['path'] + ' ' + cmd
 
+        extra_inputs_ = container.get("extra-input",[])
+        # TODO: Can't use EnsureListOf(str) yet as it handles strings as iterables.
+        # See this PR: https://github.com/datalad/datalad/pull/7267
+        # get_all=True still returns a string for a single value
+        extra_inputs_ = [extra_inputs_] if isinstance(extra_inputs_, str) else extra_inputs_
+        extra_inputs = []
+        for extra_input in set(extra_inputs_):
+            try:
+                xi_kwargs = dict(
+                    img_dspath=image_dspath,
+                    img_dirpath=op.dirname(image_path) or ".",
+                )
+                extra_inputs.append(extra_input.format(**xi_kwargs))
+            except KeyError as exc:
+                yield get_status_dict(
+                    'run',
+                    ds=ds,
+                    status='error',
+                    message=(
+                        'Unrecognized extra_input placeholder: %s. '
+                        'See containers-add for information on known ones: %s',
+                        exc,
+                        ", ".join(xi_kwargs)))
+                return
+
+        lgr.debug("extra_inputs = {!r}".format(extra_inputs))
+
         with patch.dict('os.environ',
                         {CONTAINER_NAME_ENVVAR: container['name']}):
             # fire!
             for r in run_command(
                     cmd=cmd,
                     dataset=dataset or (ds if ds.path == pwd else None),
                     inputs=inputs,
-                    extra_inputs=[image_path],
+                    extra_inputs=[image_path] + (extra_inputs or []),
```

Comment:
if I got indentation right, we always guaranteed to have `extra_inputs`  initialized to `[]` so may be just 

```suggestion
                    extra_inputs=[image_path] + extra_inputs,
```